starter-code/scripts/article.js

```diff
@@ -47,16 +47,26 @@ Article.loadAll = function(rawData) {
 // and process it, then hand off control to the View.
 Article.fetchAll = function() {
   if (localStorage.rawData) {
+    //.parse convers json we are passing through localStorage.rawData//
+    Article.loadAll(JSON.parse(localStorage.rawData));
+    // function that appends to the dom//
+    articleView.initIndexPage();
     // When rawData is already in localStorage,
     // we can load it with the .loadAll function above,
     // and then render the index page (using the proper method on the articleView object).
-    Article.loadAll(); //TODO: What do we pass in to loadAll()?
+    Article.loadAll(localStorage.rawData); //TODO: What do we pass in to loadAll()?
     //TODO: What method do we call to render the index page?
   } else {
     // TODO: When we don't already have the rawData,
     // we need to retrieve the JSON file from the server with AJAX (which jQuery method is best for this?),
     // cache it in localStorage so we can skip the server call next time,
     // then load all the data into Article.all with the .loadAll function above,
     // and then render the index page.
+    // getting the code from the server //
+    $.getJSON('data/hackerIpsum.json').then(function(rawData){
```

Comment:
Looking back at it.  you'll want to change function(rawData), to function(data), because data is what you are passing into the function and rawData is the name you are calling the data when you set it in local storage. 

You'll also what to change Article.loadAll(rawData), to Article.loadAll(data) for the same reason as above